src/infrastructure/services/auth.service.ts

```diff
@@ -0,0 +1,43 @@
+import { HttpException, HttpStatus, Injectable, Logger } from '@nestjs/common';
+import { JwtPayload, LoginStatus, LoginUserDto } from '../dto/Auth.dto';
+import { CommandBus } from '@nestjs/cqrs';
+import { JwtService } from '@nestjs/jwt';
+
+import { LoginUserCommand } from '../../domains/User/Commands/LoginUser.command';
+import { UserRepository } from '../../domains/User/User.repository';
+import { User } from '../../domains/User/User.aggregate';
+
+@Injectable()
+export class AuthService {
+  private logger = new Logger(AuthService.name);
+
+  constructor(
+    private readonly commandBus: CommandBus,
+    private userRepository: UserRepository,
+    private readonly jwtService: JwtService,
+  ) {}
+
+  async login({ username, password }: LoginUserDto): Promise<LoginStatus> {
+    const user = await this.commandBus.execute(
+      new LoginUserCommand(username, password, this.jwtService),
+    );
+
+    this.logger.debug(`logged in user: ${JSON.stringify(user)}`);
+
+    return {
+      username: user.userName.value,
+      accessToken: user.loginToken,
```

Comment:
Just sign token here, no need to do it in aggregate and store it. We dont need to read it anywhere. 
JwtStrategy takes care of extracting from Bearer and checking validity of the token. When the token is valid, only thing left to do is check if aggregate is not marked as logged out